### PR TITLE
Refactor functions in `join.py` to accept GeoSeries Input

### LINKED ISSUE #939: Convert `Point in Polygon` to Accept GeoSeries as Input

### LINKED ISSUE #940: Convert `join quadtree and bounding boxes` to Accept GeoSeries as Input

### LINKED ISSUE #941: Convert `quadtree point in polygon` to Accept GeoSeries as Input

### LINKED ISSUE #942: Convert `quadtree point to nearest linestring` to Accept GeoSeries as Input

### PR DESCRIPTION
**Pull request by isVoid**
## Description

This PR refactors all functions in `join.py` to accept GeoSeries as input.
closes #940, closes #939, closes #941, closes #942

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cuspatial/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cuspatial/cuspatial/core/spatial/join.py [hunk 1]
```diff
@@ -342,3 +323,40 @@ def quadtree_point_to_nearest_linestring(
             linestring_points_y,
         )
     )
+
+
+@cuda.jit
+def binarize(in_col, out, width):
```
Comment (harrism) [inline python/cuspatial/cuspatial/core/spatial/join.py:329]: Whoah. Shoud we move this stuff to libcuspatial?

**Review by harrism — APPROVED** [review]

File: python/cuspatial/cuspatial/core/spatial/join.py (hunk 1 above)
Comment (isVoid) [reply python/cuspatial/cuspatial/core/spatial/join.py:329]: Does this benefit cuSpatial? I don't see a usage of this other than converting the result of Point in Polygon to dataframe.

Comment (harrism) [reply python/cuspatial/cuspatial/core/spatial/join.py:329]: Doing it faster benefits cuSpatial users. Generally Numba and Python add a lot of overhead.

Comment (isVoid) [reply python/cuspatial/cuspatial/core/spatial/join.py:329]: Can probably punt to a future optimization. Another question is whether we should change the point in polygon interface, or simply leave this as a cuda kernel bind to python.

**Pull request merged**